### PR TITLE
fix(e2e): Upgrade swagger-parser to 1.0.50

### PR DESCRIPTION
**Pull request by barustum**
This change fixed a known code execution vulnerability in Swagger parser. Ran the Snyk tool before and after the change to ensure vulnerability fix.

### CODE REVIEW

File: iot-e2e-tests/edge-e2e/pom.xml [hunk 1]
```diff
@@ -21,22 +21,25 @@
             <artifactId>junit</artifactId>
             <version>4.12</version>
         </dependency>
-        <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-            <version>28.2-jre</version>
-        </dependency>
```
Comment (timtay-microsoft) [inline iot-e2e-tests/edge-e2e/pom.xml:28]: Why are you removing this dependency?

Comment (barustum) [reply iot-e2e-tests/edge-e2e/pom.xml:28]: weird, I didn't do this change explicitly. Taking a look.

**Pull request merged**